test/gtest/ucp/ucp_datatype.cc

```diff
@@ -31,6 +31,11 @@ data_type_desc_t::make(ucp_datatype_t datatype, const void *buf, size_t length,
     m_dt = datatype;
     memset(m_iov, 0, sizeof(m_iov));
 
+    if (length == 0) {
+        m_is_empty = true;
+        m_count    = 0;
+    }
```

Comment:
why need special case for empty? it can be contig

Reply:
to preserve validation interface in ucp_datatype class
(currently it assumes empty is non valid dt desc)